pkg/resource/predicates.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+// NewNameFilter returns a new *NameFilter that uses the given list.
+func NewNameFilter(list []types.NamespacedName) *NameFilter {
+	return &NameFilter{list: list}
+}
```

Comment:
That made it a lot shorter! Though i realized `controller-runtime` also had the same function and opted for that one since we'll probably not want to maintain ours after a while if they do it already.